pkg/policy/rule.go

```diff
@@ -309,6 +360,12 @@ func (r *rule) resolveIngressPolicy(ctx *SearchContext, state *traceState, resul
 			// with requirementsSelector. We don't want to modify the rule itself
 			// in the policy repository.
 			ruleCopy = *ingressRule.DeepCopy()
+			// If the rule only consists of a FromRequires statement,
+			// provide an empty selector so that it will be updated
+			// in the next step.
+			if len(ruleCopy.FromEndpoints) == 0 {
```

Comment:
I still don't understand why this is necessary - requirements rules don't explicitly allow traffic.

Reply:
If you don't do this, then it's not clear in the policy trace output what is happening when a rule with only requirements is iterated over. Let's look at this in terms of the trace output. Here's what this achieves:

Search using context `From: "any:foo"` against a rule with only `FromRequires: "any:baz"`:
```
* Rule {"matchLabels":{"any:bar":""}}: selected                                 
    Enforcing requirements [{Key:any.baz Operator:In Values:[]}]                
      No label match for [any:foo]
```

Try without this logic:
```
 * Rule {"matchLabels":{"any:bar":""}}: selected
      Found all required labels
```

...and this is quite weird. If all the required labels were found, then why doesn't this rule select and allow? It doesn't explicitly say `Allows from ....`, but it doesn't say anything else either.

Reply:
Perhaps I can gate this one also by the `ctx.Trace != TraceDisabled`.

Reply:
Makes sense, thanks!